modules/product/src/main/java/com/opengamma/strata/product/bond/ResolvedBill.java

```diff
@@ -0,0 +1,742 @@
+/*
+ * Copyright (C) 2018 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.bond;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.PropertyDefinition;
+
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.basics.date.DaysAdjustment;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.ResolvedProduct;
+import com.opengamma.strata.product.SecurityId;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import org.joda.beans.Bean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+/**
+ * A bill, resolved for pricing.
+ */
+@BeanDefinition(constructorScope = "package")
+public class ResolvedBill
+    implements ResolvedProduct, ImmutableBean, Serializable {
+
+  /**
+   * The security identifier.
+   * <p>
+   * This identifier uniquely identifies the security within the system.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final SecurityId securityId;
+  /**
+   * The currency of the bill notional.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final Currency currency;
+  /**
+   * The notional amount, must be positive.
+   * <p>
+   * The notional expressed here must be positive.
+   * The currency of the notional is specified by {@code currency}.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegativeOrZero")
+  private final double notional;
+  /**
+   * The bill maturity date.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate maturityDate;
+  /**
+   * The day count convention applicable.
+   * <p>
+   * The conversion from dates to a numerical value is made based on this day count.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final DayCount dayCount;
+  /**
+   * Yield convention.
+   * <p>
+   * The convention defines how to convert from yield to price and inversely.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final BillYieldConvention yieldConvention;
+  /**
+   * The legal entity identifier.
+   * <p>
+   * This identifier is used for the legal entity that issues the bill.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final StandardId legalEntityId;
```

Comment:
Are you ever going to need more than just the identifier? ie. a `LegalEntity` object. If so, we might want to create that now and change the bond classes while we are making backwards incompatible changes.